pulsar-io/kafka-connect-adaptor/src/main/java/org/apache/pulsar/io/kafka/connect/PulsarKafkaConnectSinkConfig.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.io.kafka.connect;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import org.apache.pulsar.io.core.annotations.FieldDoc;
+
+@Data
+@Accessors(chain = true)
+public class PulsarKafkaConnectSinkConfig implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    @FieldDoc(
+            defaultValue = "1",
+            help = "The batch size that Kafka producer will attempt to batch records together.")
+    private int batchSize = 1;
```

Comment:
why batchSize is 1?

Reply:
In kafka producer the batch size is 16384 *bytes* by default. 
Here it is number of messages the sink processes before flush. It is safer to set it to 1 and let user adjust it based on the VM size/type of kafka connect sink etc to avoid OOM or long delays between flushes.
I don't serialize the data before passing it to the kafka connect sink, and .getMessage() (that can get byte[]) is 1. optional and 2. potentially copies data internally.
I'll adjust the help message.

Reply:
1) if `batchSize` is used for representing number of messages, it should be renamed to `batchedMessages` or something else. `size` should be used for representing the total bytes. 

2) You can and probably should implement the flush logic based on bytes not messages. As you said, `messages` is not good for calculating the total amount of memory, but `bytes` can.

Reply:
@sijie this would involve doing something like `sourceRecord.getMessage().get().getData().length`
`MessageImpl.getData()` can end up allocating new array.  :/
I'll add something like getDataSize() to the Message interface (separate PR)